src/radiation/source_terms_single_group.hpp

```diff
@@ -87,7 +89,7 @@ void RadSystem<problem_t>::AddSourceTermsSingleGroup(array_t &consVar, arraycons
 
 		double coeff_n = NAN;
 		const double num_den = rho / mean_molecular_mass_;
-		if constexpr (enable_dust_gas_thermal_coupling_model_) {
+		if (enable_dust_gas_thermal_coupling_model) {
```

Comment:
Using `if` instead of `if constexpr` if it's inside of a `ParallelFor` can have a severe performance impact. This is because extra registers need to be allocated, which reduces GPU occupancy: https://quokka-astro.github.io/quokka/performance/

Reply:
This is a design/UI consideration. It might be better to allow the user to control if or not dust model is used via runtime parameter. That's way in the most recent commit I removed that parameter from the radiation_traits.

Reply:
Maybe I'm wrong. The dust/ISM model will become more complicated in the future, so we should probably make it `constexpr`.

Reply:
It can still be a runtime parameter. See this: https://github.com/AMReX-Codes/amrex/pull/2954

Reply:
> It can still be a runtime parameter. See this: [AMReX-Codes/amrex#2954](https://github.com/AMReX-Codes/amrex/pull/2954)

Interesting. I will try to implement that in a future PR, along with linear heating/cooling and photoelectric heating.